polkadot-parachains/src/chain_spec.rs

```diff
@@ -169,119 +184,78 @@ pub fn integritee_chain_spec(
-		chain_name,
-		&format!("integritee-{}", relay_chain.to_string()),
-		relay_chain.chain_type(),
-		testnet_constructor,
-		Vec::new(),
-		// telemetry endpoints
-		None,
-		// protocol id
-		Some(relay_chain.protocol_id()),
-		// fork id
-		None,
-		// properties
-		Some(
-			serde_json::from_str(
-				r#"{
-				"ss58Format": 13,
-				"tokenDecimals": 12,
-				"tokenSymbol": "TEER"
-				}"#,
-			)
-			.unwrap(),
-		),
+	#[allow(deprecated)]
+	ChainSpec::builder(
+		integritee_runtime::WASM_BINARY.expect("WASM binary was not built, please build it!"),
 		Extensions { relay_chain: relay_chain.to_string(), para_id: para_id.into() },
 	)
+	.with_name("Integritee Network")
+	.with_id(&format!("integritee-{}", relay_chain.to_string()))
+	.with_protocol_id(relay_chain.protocol_id())
+	.with_chain_type(relay_chain.chain_type())
+	.with_properties(properties)
+	.with_genesis_config_patch(integritee_genesis_config(
+		root.clone(),
+		endowed.clone(),
+		authorities.clone(),
+		para_id,
+	))
+	.build()
 }
 
 fn integritee_genesis_config(
 	root_key: AccountId,
 	endowed_accounts: Vec<AccountId>,
 	initial_authorities: Vec<AuraId>,
-	id: ParaId,
-) -> integritee_runtime::RuntimeGenesisConfig {
-	integritee_runtime::RuntimeGenesisConfig {
-		system: integritee_runtime::SystemConfig {
-			code: integritee_runtime::WASM_BINARY
-				.expect("WASM binary was not build, please build it!")
-				.to_vec(),
-			_config: Default::default(),
+	parachain_id: ParaId,
+) -> serde_json::Value {
+	serde_json::json!({
+		"aura": {
+			"authorities": initial_authorities
 		},
-		balances: integritee_runtime::BalancesConfig {
-			balances: endowed_accounts.iter().cloned().map(|k| (k, 1_000 * TEER)).collect(),
+		"balances": {
+			"balances": endowed_accounts.iter().cloned().map(|k| (k, 1_000 * TEER)).collect::<Vec<_>>(),
 		},
-		democracy: DemocracyConfig::default(),
-		council: CouncilConfig { phantom: Default::default(), members: vec![root_key.clone()] },
-		technical_committee: TechnicalCommitteeConfig {
-			phantom: Default::default(),
-			members: vec![root_key],
+		"parachainInfo": {
+			"parachainId": parachain_id,
 		},
-		vesting: Default::default(),
-		parachain_info: integritee_runtime::ParachainInfoConfig {
-			parachain_id: id,
-			_config: Default::default(),
+		"polkadotXcm": {
+			"safeXcmVersion": Some(SAFE_XCM_VERSION),
 		},
-		aura: integritee_runtime::AuraConfig { authorities: initial_authorities },
-		aura_ext: Default::default(),
-		parachain_system: Default::default(),
-		polkadot_xcm: Default::default(),
-		teerex: TeerexConfig {
-			allow_sgx_debug_mode: true,
-			allow_skipping_attestation: true,
-			_config: Default::default(),
+		"teerex": {
+			"allowSgxDebugMode": true,
+			"allowSkippingAttestation": true,
 		},
-		claims: Default::default(),
-		treasury: Default::default(),
-		transaction_payment: Default::default(),
-	}
+		"council": {
+			"members": vec![root_key.clone()]
+		},
+		"technicalcommittee": {
```

Comment:
hmm, why is this suddenly like this. It seems less typesafe too me?

Reply:
good catch
the goal is to no longer build native runtimes, so all needs to be defined in json